aws-embedded-metrics-customink.gemspec

```diff
@@ -20,4 +20,5 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
   spec.add_dependency 'concurrent-ruby'
+  spec.add_dependency 'tcp-client'
```

Comment:
I'm wondering if that can be a peer dependency similar to how ActiveRecord does this with adapters. Hence, a require would at runtime look for the dep. Example: https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb#L6-L7

Reply:
Neat, I didn't know you could do that! Let me try that out.